django-cloudlaunch/cloudlaunch/models.py

```diff
@@ -225,6 +225,16 @@ class ApplicationDeploymentTask(models.Model):
     def __str__(self):
         return "{0}".format(self.id)
 
+    def save(self, *args, **kwargs):
+        # validate at most one LAUNCH task per deployment
+        if self.action == self.LAUNCH:
+            if ApplicationDeploymentTask.objects.filter(
```

Comment:
One problem with this is that this is not atomic. I guess we should make this transactional, so that there's no race condition between checking and saving. Or we could just ignore it for now...

Reply:
Very good point, I hadn't thought of that. Thanks Nuwan.